duckduckgo/iOS
# Custom Instrumentation

Task/Issue URL: https://app.asana.com/0/856498667320406/1122248096879715
Tech Design URL:
CC:

**Description**:
Added Custom Instruments target to project. Please check README.md for instruction how to run it. Note, that this is initial implementation and functionality is limited. :)

Features:
- Info about URLs being loaded in the tabs (only successful ones are logged for now).
- Info about performance-related events like Content Blocker updates, DisconnectMe Parsing/loading, data cleaning, tab initialization.
- Info about which requests are blocked and which are not.

**Steps to test this PR**:
1. Ensure README.md looks ok.
2. Ensure that debug JS messaging is only enabled in debug builds.
3. Ensure no uninvited logs are being logged to console in release build.
4. Test and play with it in instruments.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
